src/Avalonia.Controls.ColorPicker/Themes/Fluent/ColorPicker.xaml

```diff
@@ -0,0 +1,82 @@
+﻿<Styles xmlns="https://github.com/avaloniaui"
+        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        xmlns:local="using:Avalonia.Controls"
+        xmlns:converters="using:Avalonia.Controls.Converters"
+        x:CompileBindings="False">
+
+  <Styles.Resources>
+    <converters:ToBrushConverter x:Key="ToBrush" />
+    <converters:CornerRadiusFilterConverter x:Key="LeftCornerRadiusFilterConverter" Filter="TopLeft, BottomLeft"/>
+  </Styles.Resources>
+
+  <Style Selector="ColorPicker">
+    <Setter Property="CornerRadius" Value="{DynamicResource ControlCornerRadius}" />
+    <Setter Property="Height" Value="32" />
+    <Setter Property="Width" Value="64" />
+    <Setter Property="MinWidth" Value="64" />
+    <Setter Property="Template">
+      <ControlTemplate>
+        <DropDownButton CornerRadius="{TemplateBinding CornerRadius}"
+                        Height="{TemplateBinding Height}"
+                        Width="{TemplateBinding Width}"
+                        HorizontalContentAlignment="Stretch"
+                        VerticalContentAlignment="Stretch"
+                        Padding="0,0,10,0"
+                        UseLayoutRounding="False">
+          <DropDownButton.Styles>
+            <Style Selector="FlyoutPresenter.NoPadding">
+              <Setter Property="Padding" Value="0" />
+            </Style>
+          </DropDownButton.Styles>
+          <DropDownButton.Content>
+            <!-- Preview color -->
+            <Panel>
+              <Border Background="{StaticResource CheckeredBackgroundBrush}"
+                      CornerRadius="{TemplateBinding CornerRadius, Converter={StaticResource LeftCornerRadiusFilterConverter}}"
+                      HorizontalAlignment="Stretch"
+                      VerticalAlignment="Stretch"
+                      Margin="1,1,0,1" />
+              <Border Background="{Binding HsvColor, ElementName=FlyoutColorView, Converter={StaticResource ToBrush}, Mode=OneWay}"
+                      CornerRadius="{TemplateBinding CornerRadius, Converter={StaticResource LeftCornerRadiusFilterConverter}}"
+                      HorizontalAlignment="Stretch"
+                      VerticalAlignment="Stretch"
+                      Margin="1,1,0,1" />
+            </Panel>
+          </DropDownButton.Content>
+          <DropDownButton.Flyout>
+            <Flyout FlyoutPresenterClasses="NoPadding">
+              <!-- Skip the following:
+                - CustomPaletteColors
```

Comment:
I also can't bind to CustomPaletteColors since it's a direct property and a read-only collection. I really don't want to sync the two collections manually with code-behind and event handling. However, I don't see another way. Am I missing something? Is it possible to bind in this scenario?

Note that if I allowed setting the collection it could work. But it's considered bad practice to allow setters for collecitons. Perhaps make it a get-only CLR property but a StyledProperty IEnumerable or something?

Reply:
Yes, as you said it's a readonly property (both CLR and DirectProp definition), and XAML doesn't know how to bind it. It doesn't know that ObservableCollection can do it.

I don't know other way, but use some help from code behind. Not exactly the same situation, but we did it for Flyout.PresenterClasses too - https://github.com/AvaloniaUI/Avalonia/blob/e51851e68f8ed93544713f428775113998a45a4e/src/Avalonia.Controls/Flyouts/FlyoutBase.cs#L564

Reply:
Is this kind of bindings supported in WPF/UWP? I don't remember if so, but I wish it could.

Reply:
> Is this kind of bindings supported in WPF/UWP? I don't remember if so, but I wish it could.

In UWP this certainly isn't supported. In WPF I also don't think so and have never seen it done. It's the one common case that has just never been handled correctly by any dependency property system...

Ideally, the DP system needs to special-case collections. A supported collection type could implement a new IApplyTo interface or something that would allow copying items from one collection into another by reference. While the interface would be easy I imagine the property system updates are not. This also hasn't been fully thought out so who knows what other issues there are. Anyway, I found something else.

> Not exactly the same situation, but we did it for Flyout.PresenterClasses too -

Thanks for the example.

I also found some cases where IEnumerable is used that seems to work-around this issue (taken from ItemsControl). I haven't seen done before in other XAML frameworks. Perhaps this is how it should be done in Avalonia?

https://github.com/AvaloniaUI/Avalonia/blob/56ef3adf065aba960648bb85466d03651583bce8/src/Avalonia.Controls/ItemsControl.cs#L105-L109

 (note it is still bad convention to have a setter for a collection/enumerable... but maybe that's the lesser of two evils to get this to work in the property system as-is)

Reply:
Now I think that public setter for a collections is not that uncommon in avalonia. Transitions is the same.

Reply:
If that is what is done in Avalonia I'll do it to get this to work. It just seems strange to me based on other XAML experience.

I still consider this an unsolved problem in the binding system though and it should be special cased.

Reply:
Switched to a StyledProperty following `ItemsControl`.

```csharp
        public static readonly StyledProperty<IEnumerable<Color>?> PaletteColorsProperty =
            AvaloniaProperty.Register<ColorView, IEnumerable<Color>?>(
                nameof(PaletteColors),
                null);

        public IEnumerable<Color>? PaletteColors
        {
            get => GetValue(PaletteColorsProperty);
            set => SetValue(PaletteColorsProperty, value);
        }
```